--- a/src/unit_test.c
+++ b/src/unit_test.c
@@ -952,7 +952,11 @@
     for(c = 0 ; c < max ; c++) {
         calculated_number v = unpack_storage_number(rd->values[c]);
         calculated_number n = test->results[c];
+#ifdef __UCLIBC__
+        int same = (llround(v * 10000000.0) == llround(n * 10000000.0))?1:0;
+#else
         int same = (roundl(v * 10000000.0) == roundl(n * 10000000.0))?1:0;
+#endif
         fprintf(stderr, "    %s/%s: checking position %lu (at %lu secs), expecting value " CALCULATED_NUMBER_FORMAT ", found " CALCULATED_NUMBER_FORMAT ", %s\n",
             test->name, rd->name, c+1,
             (rrdset_first_entry_t(st) + c * st->update_every) - time_start,
@@ -963,7 +967,11 @@
         if(rd2) {
             v = unpack_storage_number(rd2->values[c]);
             n = test->results2[c];
+#ifdef __UCLIBC__
+            same = (llround(v * 10000000.0) == llround(n * 10000000.0))?1:0;
+#else
             same = (roundl(v * 10000000.0) == roundl(n * 10000000.0))?1:0;
+#endif
             fprintf(stderr, "    %s/%s: checking position %lu (at %lu secs), expecting value " CALCULATED_NUMBER_FORMAT ", found " CALCULATED_NUMBER_FORMAT ", %s\n",
                 test->name, rd2->name, c+1,
                 (rrdset_first_entry_t(st) + c * st->update_every) - time_start,
--- a/src/rrd2json.c
+++ c/src/rrd2json.c
@@ -286,14 +286,22 @@
                         buffer_sprintf(wb, "NETDATA_%s_%s=\"\"      # %s\n", chart, dimension, st->units);
                     else {
                         if(rd->multiplier < 0 || rd->divisor < 0) n = -n;
+#ifdef __UCLIBC__
+                        n = llround(n);
+#else
                         n = roundl(n);
+#endif
                         if(!rrddim_flag_check(rd, RRDDIM_FLAG_HIDDEN)) total += n;
                         buffer_sprintf(wb, "NETDATA_%s_%s=\"%0.0Lf\"      # %s\n", chart, dimension, n, st->units);
                     }
                 }
             }
 
+#ifdef __UCLIBC__
+            total = llround(total);
+#else
             total = roundl(total);
+#endif
             buffer_sprintf(wb, "NETDATA_%s_VISIBLETOTAL=\"%0.0Lf\"      # %s\n", chart, total, st->units);
             rrdset_unlock(st);
         }
@@ -316,7 +324,11 @@
         if(isnan(n) || isinf(n))
             buffer_sprintf(wb, "NETDATA_ALARM_%s_%s_VALUE=\"\"      # %s\n", chart, alarm, rc->units);
         else {
+#ifdef __UCLIBC__
+            n = llround(n);
+#else
             n = roundl(n);
+#endif
             buffer_sprintf(wb, "NETDATA_ALARM_%s_%s_VALUE=\"%0.0Lf\"      # %s\n", chart, alarm, n, rc->units);
         }
 
